### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by sahlminkok**
### In this Pull Request I:
- Create Person Class
- Create Student Class
- Create Teacher Class

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(_name, age, _parent_permission, classroom)
+    super('Unknown', age, true)
```
Comment (OyePriscilla) [inline student.rb]: - [ ] Please kindly declare the parameters inherited from the `Person.rb` appropriaely as you did in `Person.rb` file. The `super()` function is used to call methods and constructors of the parent class, kindly ensure that the parameters are written correctly. 👍

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(_name, age, specialization)
+    super('Unknown', age)
```
Comment (OyePriscilla) [inline teacher.rb]: - [ ] Please kindly declare the parameters inherited from the `Person.rb` appropriaely as you did in `Person.rb` file. The `super()` function is used to call methods and constructors of the parent class, kindly ensure that the parameters are written correctly. 👍 **Note: parent_permission=true is missing from the constructor as well**

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @sahlminkok 👍 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented individual `.rb` file for each class (`Person`, `Student`, and `Teacher` ) ✔️ 
Implemented inheritance with Ruby ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @sahlminkok,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**